src/Nethermind/Nethermind.Runner/Program.cs

```diff
@@ -591,4 +593,57 @@ private static string GetProductInfo()
 
         return info.ToString();
     }
+
+    private static HashSet<string> generateValidConfigOptionsHash()
+    {
+        var options = new HashSet<string>();
+
+        Type configurationType = typeof(IConfig);
+        IEnumerable<Type> configTypes = TypeDiscovery.FindNethermindBasedTypes(configurationType)
+            .Where(ct => ct.IsInterface);
+
+        foreach (Type configType in configTypes.Where(ct => !ct.IsAssignableTo(typeof(INoCategoryConfig))).OrderBy(c => c.Name))
+        {
+            if (configType is null)
+            {
+                continue;
+            }
+
+            ConfigCategoryAttribute? typeLevel = configType.GetCustomAttribute<ConfigCategoryAttribute>();
+
+            if (typeLevel is not null && (typeLevel?.DisabledForCli ?? true))
+            {
+                continue;
+            }
+
+            foreach (PropertyInfo propertyInfo in configType
+                            .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                            .OrderBy(p => p.Name))
+            {
+                ConfigItemAttribute? configItemAttribute = propertyInfo.GetCustomAttribute<ConfigItemAttribute>();
+                if (!(configItemAttribute?.DisabledForCli ?? false))
+                {
+                    options.Add($"--{configType.Name[1..].Replace("Config", string.Empty)}.{propertyInfo.Name}");
+                }
+            }
+        }
+
+        return options;
+    }
+
+    private static void ValidateCommandLineFlags(string[] args)
```

Comment:
If we are validating here I would also move `GetDuplicateArguments` and duplication validation under this one method.